### DISCUSSION TITLE
cellの値の受け渡し

### DISCUSSION THREAD

**Discussion by Keita0920** [lit-iphone-course/lit-iphone-course#58] [Question-original] [answered]
(1) 実現したいこと
cellを選択するとtagListViewControllerからtagAddViewControllerに遷移して、遷移先のtextfieldにcell内のデータを表示したい。

(2) 実現するために自分が試した内容とその結果
tagListViewController内でインスタンスを作ってcellの中身を代入するとoutlet接続の変数がnilのエラーになってしまう

(3) 調べた内容や資料
1.https://teratail.com/questions/215155
2.https://note.com/engineer_memo/n/ne5b6689e8afb
3.https://capibara1969.com/634/

(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
tagListViewController内
func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let tagAddVC = storyboard?.instantiateViewController(withIdentifier: "TagAddViewController") as? TagAddViewController
        if let tagAddVC = tagAddVC {
            tagAddVC.receiveTagName = self.tagNameArray[(self.table.indexPathForSelectedRow?.row)!]
            tagAddVC.tagName.text=tagAddVC.receiveTagName
        }
            // セルの選択を解除
            tableView.deselectRow(at: indexPath, animated: true)
            // 別の画面に遷移
            performSegue(withIdentifier: "editTag", sender: nil)
        }

(5) GitHubのリンク
https://github.com/Keita0920/LA_RePlace

**Comment by appleEmily** [comment, marked as answer]
<img width="674" alt="image" src="https://user-images.githubusercontent.com/49332380/192244407-40287ccb-9d7b-48b1-a5ea-6372f2863fcb.png">
tableViewのcellをタップした時の値渡しについては、添付画像のようにindexPath.rowを使っていただければいけます！

で、それだけだとアプリは落ちなくなるんだけど、画面遷移してもtextFieldには何も値は入っていません。
その理由は、「画面遷移の仕方」に問題があるからです。今、次のVCはTagAddViewControllerです〜っていうのをTagListVCの56行目らへんで書いてて、そこで値も渡しているのだけど、画面遷移自体はsegueで行ってます。二つの方法の画面遷移を混ぜてしまっているのでうまくいきません。
どっちかに統一してあげてください〜
そして、せっかくnavigationControllerを使っているので、navigationControllerを使った画面遷移・値渡しをしてあげれば良いです！

navigationControllerを使って画面遷移するコードはこんな感じ
`self.navigationController?.pushViewController(tagAddVC!, animated: true)`
[参考になる記事](https://www.macky-studio.com/entry/2019/04/24/020247) これ以外にも検索すればたくさん出てくる！
値渡しは今書いてくれている、modalでの遷移と同じ感じで行けます〜

**Reply by Keita0920** [reply]
できました！ありがとうございます！
